### PR TITLE
Fix polkadot extrinsic result parsing

### LINKED ISSUE #4544: We are not fetching extrinsic result properly for polkadot
**Describe the bug**

Offending code (my fault :-/):

```
function getExtrinsicSuccess(index, blockEvents) {
  blockEvents.forEach((record) => {
    const { event, phase } = record
    if (
      parseInt(phase.toHuman().ApplyExtrinsic) === index &&
      event.section === `system` &&
      event.method === `ExtrinsicFailed`
    ) {
      return false
    }
  })
  return true
}
```

Problems:

- ForEach is not able to return anything, so this always return true!
- Better search for an ExtrinsicSuccess event as it's more common than ExtrinsicFailed and use a `for` loop to take advantage of early loop termination.

So, this should be:

function getExtrinsicSuccess(index, blockEvents) {

```
function getExtrinsicSuccess(extrinsicIndex, blockEvents) {
  let extrinsicSuccess = false
  for (let index = 0, length = blockEvents.length; index < length; i++) {
    const { event, phase } = blockEvents[index]
    if (
      parseInt(phase.toHuman().ApplyExtrinsic) === extrinsicIndex &&
        event.section === `system` &&
        event.method === `ExtrinsicSuccess`
    ) {
      extrinsicSuccess = true
      break
    }
  }
  return extrinsicSuccess
}
```

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: 
 - Browser: 
 - Version: 

**Smartphone (please complete the following information):**
 - Device: 
 - OS: 
 - Browser: 
 - Version: 

**Additional context**

### PR DESCRIPTION
**Pull request by mariopino**
Closes #4544 

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: api/lib/reducers/polkadotV0-reducers.js [hunk 1]
```diff
@@ -241,18 +241,20 @@ function parsePolkadotTransaction(
...
+function getExtrinsicSuccess(extrinsicIndex, blockEvents) {
+  let extrinsicSuccess = false
+  for (let index = 0, length = blockEvents.length; index < length; i++) {
+    const { event, phase } = blockEvents[index]
     if (
-      parseInt(phase.toHuman().ApplyExtrinsic) === index &&
-      event.section === `system` &&
-      event.method === `ExtrinsicFailed`
+      parseInt(phase.toHuman().ApplyExtrinsic) === extrinsicIndex &&
+        event.section === `system` &&
```
Comment (Bitcoinera) [inline api/lib/reducers/polkadotV0-reducers.js:250]: can you please lint this file?

**Comment by Bitcoinera** [comment]
I cannot test this because Westend is broken xD

**Comment by Bitcoinera** [comment]
image

:eyes:

File: api/lib/reducers/polkadotV0-reducers.js [hunk 2]
```diff
@@ -241,18 +241,20 @@ function parsePolkadotTransaction(
   }
 }
 
-function getExtrinsicSuccess(index, blockEvents) {
-  blockEvents.forEach((record) => {
-    const { event, phase } = record
+function getExtrinsicSuccess(extrinsicIndex, blockEvents) {
+  let extrinsicSuccess = false
+  for (let index = 0, length = blockEvents.length; index < length; index++) {
```
Comment (faboweb) [inline api/lib/reducers/polkadotV0-reducers.js:246]:
```suggestion
  blockEvents.forEach(({ event, phase })) => {
```

File: api/lib/reducers/polkadotV0-reducers.js (hunk 1 above)
Comment (mariopino) [reply api/lib/reducers/polkadotV0-reducers.js]: Done!

**Comment by mariopino** [comment]
> I cannot test this because Westend is broken xD

I used https://polkadot.js.org/apps/#/js to test it, just search for a failed/ok tx and use this code:

```
const blockEvents = await 
// https://v3dev2.polkastats.io/block/?blockNumber=1393893 include a failed tx
api.query.system.events.at("0xad6b996eff382df11343174622e5ef0866b582be2c99cb1432ac4997f41fb17f");
console.log(getExtrinsicSuccess(2, blockEvents));

function getExtrinsicSuccess(extrinsicIndex, blockEvents) {
  return blockEvents.find(
    ({ event, phase }) =>
      parseInt(phase.toHuman().ApplyExtrinsic) === extrinsicIndex &&
      event.section === `system` &&
      event.method === `ExtrinsicSuccess`
  )
    ? true
    : false
}
```

**Pull request merged**